code/modules/reagents/reagent_containers/hypospray.dm

```diff
@@ -16,141 +16,101 @@
 	var/infinite = FALSE
 	var/has_cap = TRUE
 
+
 /obj/item/reagent_containers/hypospray/attack(mob/living/M, mob/user)
 	if(has_cap)
 		to_chat(user, span_warning("[src] has a cap on! You need to remove it first."))
 		return FALSE
-		playsound(src, 'modular/Smoker/sound/inject.ogg')
 	inject(M, user)
+	..()
+
+/obj/item/reagent_containers/hypospray/medipen/sealbottle/inject(mob/living/M, mob/user)
+	return ..(M, user, src, drinking = TRUE)  // Call parent, but mark as drinking
 
-///Handles all injection checks, injection and logging.
-/obj/item/reagent_containers/hypospray/proc/inject(mob/living/M, mob/user)
-	if(!reagents.total_volume)
-		to_chat(user, span_warning("[src] is empty!"))
+/obj/item/reagent_containers/hypospray/medipen/sty/inject(mob/living/M, mob/user)
+	return ..(M, user, src, drinking = FALSE)  // Call parent with normal injection behavior
```

Comment:
I don't think this override is necessary at all? `drinking = FALSE` is the default.

Reply:
Yeah, You're correct.